### PR TITLE
*: Table partition double write during Reorganize partition (part 2) | tidb-test=pr/2044

### PR DESCRIPTION
**Pull request by mjonss**
### What problem does this PR solve?

Issue Number: ref #15000 , ref #38535

Problem Summary:

- **Only this!** Adding double write during DDL for reorganize partition.
- No tests, https://github.com/pingcap/tidb/pull/38532 includes both https://github.com/pingcap/tidb/pull/38460 and this PR  and adds the rest of the DDL code so it is complete and can be tested (part 3 also includes test for this PR).

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: executor/builder.go [hunk 1]
```diff
@@ -3527,10 +3527,14 @@ func getPartitionKeyColOffsets(keyColIDs []int64, pt table.PartitionedTable) []i
 		keyColOffsets[i] = offset
 	}
 
-	pe, err := pt.(interface {
-		PartitionExpr() (*tables.PartitionExpr, error)
-	}).PartitionExpr()
-	if err != nil {
+	t, ok := pt.(interface {
```
Comment (bb7133) [inline executor/builder.go:3530]: (For now)Is it possible that `ok` is false?

Comment (mjonss) [reply executor/builder.go:3530]: No, it should not be possible currently, but I think it is better to check anyway :)

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**